### PR TITLE
Enable IProfiling While a Method is in the Compilation Queue

### PR DESCRIPTION
**Pull request by luke-li-2003**
Allow the IProfiler to run on methods while they are inside the compilation queues for better throughput

### CODE REVIEW

**Comment by luke-li-2003** [comment]
@mpirvu

**Review by mpirvu — APPROVED** [review]
LGTM. I would like to validate the effect on something like Daytrader before merging this

**Comment by mpirvu** [comment]
jenkins test sanity all jdk21

**Comment by mpirvu** [comment]
I performed a series of performance experiments with this change and I am documenting here the results:

Daytrader8, 4P mode: no change in throughput.
AcmeAirEE8, 4P mode: 1.1% improvement in throughput(CI95=0.6%)
AcmeAirEE8, 1P mode: no change in throughput. Possible 1.8% increase in footprint
Ilog, 5 ruleset, 4P mode: 0.9% throughput improvement; could be noise (CI95=0.8%)
Ilog, 300 ruleset, 4P mode: 1.7% throughput improvement; could be noise (CI95=1.5%)
DT8 startup-time, 4P mode: no change in start-up time for warm runs. 18% longer start-up time for cold runs.

To summarize: the change may improve throughput a little for some applications. One worrisome aspect is the startup time for cold runs which takes a large hit.
